### ISSUE TITLE
Python REPL

### ISSUE THREAD

**Issue by andybak** [antoniomdk/Unity-Python#1]
Would it be feasible to embed IPython?

https://stackoverflow.com/questions/1849652/how-to-use-ipython-with-ironpython

Any useful REPL would need history and basic autocompletion so reusing a mature project would give a lot of value.

I know nothing about Unity's support for .NET embedding however. It wouldn't have to necessarily integrate into the UI neatly however - that is less critical than having the features of a good shell.

**Comment by antoniomdk** [comment]
Remember that IronPython version is 2.6.2 which is the only one compatible with .NET 2.0 (Unity version of .net).

**Comment by andybak** [comment]
I didn't know that. I'm not especially familiar with the .Net ecosystem (I'm only here because of Unity).

So - does that rule out IPython entirely?

**Comment by antoniomdk** [comment]
Yes, totally. Altought it is posible to create an interactive shell inside the editor and execute it in runtime. Even an editor in-game/app and execute code.

**Comment by andybak** [comment]
We don't need the full feature set of IPython. I'd say the following would be essential features:

1. Shell history and Ctrl+R history search
2. Syntax coloring
3. Some level of Tab autocomplete.

Do you know of a shell that works in 2.6.2 that offers any of this?

**Comment by antoniomdk** [comment]
I think, there is no shell included in Iron Python for that version. But even if it exists, it could not be used as is in Unity.

**Comment by antoniomdk** [comment]
Now, Unity supports .NET 4.6. It is possible to implement this. I'm going to work on it.
